File: src/mobile/models/ride.ts
export interface RideLocation {
  latitude: number;
  longitude: number;
  timestamp?: number;
  elevation?: number;
  speed?: number;
}

export default class Ride {
  id?: string;
  name?: string;
  description?: string;
  distance?: number;
  duration?: number;
  averageSpeed?: number;
  totalElevation?: number;
  minSpeed?: number;
  maxSpeed?: number;
  minElevation?: number;
  maxElevation?: number;
  elevation?: number;
  createdById?: string;
  locations?: RideLocation[];
}

File: src/api/Entities/Ride.cs
using System.ComponentModel.DataAnnotations;

namespace BikerHub.Entities;

public class RideEntity : BaseEntity<Guid>
{

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }
    public decimal Distance { get; set; }
    public decimal Duration { get; set; }
    public decimal AverageSpeed { get; set; }
    public decimal TotalElevation { get; set; }
    public decimal MinSpeed { get; set; }
    public decimal MaxSpeed { get; set; }
    public decimal MinElevation { get; set; }
    public decimal MaxElevation { get; set; }
    public string LocationsJson { get; set; } = string.Empty;
}
